web/src/components/BlockList.vue

```diff
@@ -0,0 +1,293 @@
+                    :block="selectedBlock"
+                    class="no-y-padding"
+                  >
+                  </BlockVisualizer>
+                  <!-- this second blockVisualizer makes sure that this component does not resize whenever a -->
+                  <!-- block is selected/deselected -->
+                  <BlockVisualizer
+                    v-else
+                    :block="diff.blocks[0]"
+                    :dummy="true"
+                    class="no-y-padding"
+                  >
+                  </BlockVisualizer>
+                </v-col>
+                <v-col cols="auto">
+                  <v-btn @click.stop="changeSelectedItem(1)" ref="buttonright1">
+                    <v-icon>
+                      mdi-arrow-right-thick
+                    </v-icon>
+                  </v-btn>
+                </v-col>
+              </v-row>
+            </v-col>
+          </v-row>
+          <v-row>
+            <v-container class="no-y-padding" fluid>
+              <v-row>
+                <v-col class="no-y-padding" cols="12">
+                  <v-list-item-subtitle>
+                    Minimum block length
+                  </v-list-item-subtitle>
+                  <v-slider
+                    :max="highestBlockLength + 1"
+                    :min="lowestBlockLength"
+                    @end="applyMinBlockLength"
+                    thumb-label
+                    track-color="lightgray"
+                  >
+                  </v-slider>
+                </v-col>
+              </v-row>
+            </v-container>
+          </v-row>
+        </v-container>
+      </v-col>
+    </v-row>
+    <v-divider></v-divider>
+    <v-row no-gutters>
+      <v-data-table
+        :headers="headers"
+        :item-class="itemClassFunction"
+        :items="blocksWithId"
+        @click:row="onRowClick"
+        disable-pagination
+        fixed-header
+        height="71vh"
+        hide-default-footer
+        id="blockList"
+        item-key="id"
+        multi-sort
+        selectable-key="active"
+        single-select
+        style="width: 100%"
+        v-model="dataTableSelection"
+      >
+        <template v-slot:item.active="{ item }">
+          <v-simple-checkbox
+            :ripple="false"
+            @input="dataTableCheckBoxToggle(item, $event)"
+            color="primary"
+            off-icon="mdi-eye-off"
+            on-icon="mdi-eye"
+            v-model="item.active"
+          ></v-simple-checkbox>
+        </template>
+      </v-data-table>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import BlockVisualizer from "@/components/BlockVisualizer.vue";
+import { Block, Diff } from "@/api/api";
+import { Component, Prop, PropSync, Vue, Watch } from "vue-property-decorator";
+import { constructID, SelectionId } from "@/util/OccurenceHighlight";
+import { SideID } from "@/components/CompareCard.vue";
+
+interface BlockWithId extends Block {
```

Comment:
We could add an id to every block?

Reply:
I thnk blocks already have an id in the CLI, maybe this should be propagated to the frontend as well?

Reply:
That is possible however in my code the id is assumed to be the index of the block in the list present within the diff. Maybe I should rename `id` to `index` to avoid confusion?